contracts/contracts/strategies/NativeStaking/ValidatorAccountant.sol

```diff
@@ -0,0 +1,261 @@
+        uint256 noOfValidators,
+        uint256 remainingValidators,
+        uint256 wethSentToVault
+    );
+    event AccountingValidatorSlashed(
+        uint256 remainingValidators,
+        uint256 wethSentToVault
+    );
+    event AccountingConsensusRewards(uint256 amount);
+
+    event AccountingManuallyFixed(
+        int256 validatorsDelta,
+        int256 consensusRewardsDelta,
+        uint256 wethToVault
+    );
+
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _vaultAddress Address of the Vault
+    /// @param _beaconChainDepositContract Address of the beacon chain deposit contract
+    /// @param _ssvNetwork Address of the SSV Network contract
+    /// @param _maxValidators Maximum number of validators that can be registered in the strategy
+    constructor(
+        address _wethAddress,
+        address _vaultAddress,
+        address _beaconChainDepositContract,
+        address _ssvNetwork,
+        uint256 _maxValidators
+    )
+        ValidatorRegistrator(
+            _wethAddress,
+            _vaultAddress,
+            _beaconChainDepositContract,
+            _ssvNetwork,
+            _maxValidators
+        )
+    {}
+
+    /// @notice set fuse interval values
+    function setFuseInterval(
+        uint256 _fuseIntervalStart,
+        uint256 _fuseIntervalEnd
+    ) external onlyGovernor {
+        require(
+            _fuseIntervalStart < _fuseIntervalEnd &&
+                _fuseIntervalEnd < 32 ether &&
+                _fuseIntervalEnd - _fuseIntervalStart >= 4 ether,
+            "Incorrect fuse interval"
+        );
+
+        fuseIntervalStart = _fuseIntervalStart;
+        fuseIntervalEnd = _fuseIntervalEnd;
+
+        emit FuseIntervalUpdated(_fuseIntervalStart, _fuseIntervalEnd);
+    }
+
+    /* solhint-disable max-line-length */
+    /// This notion page offers a good explanation of how the accounting functions
+    /// https://www.notion.so/originprotocol/Limited-simplified-native-staking-accounting-67a217c8420d40678eb943b9da0ee77d
+    /// In short, after dividing by 32, if the ETH remaining on the contract falls between 0 and fuseIntervalStart,
+    /// the accounting function will treat that ETH as Beacon chain consensus rewards.
+    /// On the contrary, if after dividing by 32, the ETH remaining on the contract falls between fuseIntervalEnd and 32,
+    /// the accounting function will treat that as a validator slashing.
+    /// @notice Perform the accounting attributing beacon chain ETH to either full or partial withdrawals. Returns true when
+    /// accounting is valid and fuse isn't "blown". Returns false when fuse is blown.
+    /// @dev This function could in theory be permission-less but lets allow only the Registrator (Defender Action) to call it
+    /// for now.
+    /// @return accountingValid true if accounting was successful, false if fuse is blown
+    /* solhint-enable max-line-length */
+    function doAccounting()
```

Comment:
Is there any need for the strategist to call this, or should the strategist just call manuallyFixAccounting with zero values to force an accounting?

Reply:
Good question. `manuallyFixAccounting` requires the strategy to be paused, but the Strategist can `pause` the contract, call `manuallyFixAccounting` which then calls `doAccounting`. I think that is enough.